packages/astro/src/vite-plugin-mdx/tag.ts

```diff
@@ -9,115 +12,100 @@ import * as t from '@babel/types';
  * This plugin crawls each export in the file and "tags" each export with a given `rendererName`.
  * This allows us to automatically match a component to a renderer and skip the usual `check()` calls.
  */
-export default async function tagExportsWithRenderer({
-	rendererName,
-}: {
-	rendererName: string;
-	root: URL;
-}): Promise<PluginObj> {
-	return {
-		visitor: {
-			Program: {
-				// Inject `import { __astro_tag_component__ } from 'astro/runtime/server/index.js'`
-				enter(path) {
-					path.node.body.splice(
-						0,
-						0,
-						t.importDeclaration(
-							[
-								t.importSpecifier(
-									t.identifier('__astro_tag_component__'),
-									t.identifier('__astro_tag_component__')
-								),
-							],
-							t.stringLiteral('astro/runtime/server/index.js')
-						)
-					);
-				},
-				// For each export we found, inject `__astro_tag_component__(exportName, rendererName)`
-				exit(path, state) {
-					const exportedIds = state.get('astro:tags');
-					if (exportedIds) {
-						for (const id of exportedIds) {
-							path.node.body.push(
-								t.expressionStatement(
-									t.callExpression(t.identifier('__astro_tag_component__'), [
-										t.identifier(id),
-										t.stringLiteral(rendererName),
-									])
-								)
-							);
-						}
-					}
-				},
+export const tagExportsPlugin: PluginObj = {
```

Comment:
I converted the function to a simple object so the instance can be shared. Nothing else within the function body is changed.